src/Servers/Kestrel/Core/src/Internal/Http2/HPack/IntegerEncoder.cs

```diff
@@ -36,17 +36,18 @@ public static bool Encode(int i, int n, Span<byte> buffer, out int length)
                     return false;
                 }
 
-                i = i - ((1 << n) - 1);
+                i -= ((1 << n) - 1);
                 while (i >= 128)
                 {
-                    buffer[j++] = (byte)(i % 128 + 128);
+                    var ui = (uint)i; // Use unsigned for optimizations
+                    buffer[j++] = (byte)((ui & 127) + 128); // (i % 128 + 128)
```

Comment:
```suggestion
                    buffer[j++] = (byte)((ui % 128) + 128);
```
For `uint` the JIT will perform the bit-hack.
Sorry, I was to unclear with my suggestion.